environment

```diff
@@ -2,7 +2,7 @@
 #
 # Copyright (c) 2019-2023 Jeremy Green.
 
-SOURCE_DIR="$(readlink -f "$(dirname "${BASH_SOURCE[0]}")")"
+SOURCE_DIR="$(dirname "$(readlink -f "${BASH_SOURCE[0]}")")"
```

Comment:
This was back to front, so the symlink wasn't resolved correctly when sourcing from src/frontend.